R/addWaterYear.R

```diff
@@ -0,0 +1,83 @@
+#' add a water year column
+#' 
+#' Add a column to the dataRetrieval data frame with the water year. WQP 
+#' queries will return a water year column for the start and end dates
+#' of the data.
+#' 
+#' @param rawData the daily- or unit-values datset retrieved from NWISweb. Must
+#' have at least one of the following columns to add the new water year columns:
+#' `dateTime`, `Date`, `ActivityStartDate`, or `ActivityEndDate`. The date column(s)
+#' can be character, POSIXct, Date. They cannot be numeric.
+#' 
+#' @return data.frame with an additional integer column with "WY" appended to the 
+#' date column name. For WQP, there will be 2 columns: `ActivityStartDateWY` and 
+#' `ActivityEndDateWY`.
+#' @export
+#' 
+#' @importFrom dplyr select
+#' @importFrom dplyr everything
+#' @examples
+#' \dontrun{ 
+#' dataTemp <- readNWISdata(stateCd="OH",parameterCd="00010", service="dv")
+#' dataTemp <- addWaterYear(dataTemp)
+#' 
+#' pHData <- readWQPdata(siteid="USGS-04024315",characteristicName="pH")
+#' pHData <- addWaterYear(pHData)
+#' }
+addWaterYear <- function(rawData){
+  
+  allowedDateColNames <- c("dateTime", "Date", "ActivityStartDate", "ActivityEndDate")
+  
+  # only allow WY to be added if there is an appropriate date column
+  if(all(!allowedDateColNames %in% names(rawData))){
+    stop("specified date column does not exist in supplied data frame")
+  }
+  
+  # set the name of the date column(s) to use for calculating the WY
+  dateColName <- names(rawData)[names(rawData) %in% allowedDateColNames]
+  
+  # set up new df that will include water year columns
+  wyData <- rawData
+  
+  for(dateCol in dateColName){
+    dateColWY <- paste0(dateCol, "WY")
+    
+    # if this WY column already exists, do not add another
+    if(dateColWY %in% names(wyData)){
+      next
+    }
+    
+    # calculate WY & add as new column
+    wyData[[dateColWY]] <- calcWaterYear(wyData[[dateCol]])
+    
+    # move waterYear so that it is always comes right after dateTime
+    dateCol_i <- which(names(wyData) == dateCol)
+    dateColWY_i <- which(names(wyData) == dateColWY)
+    wyData <- select(wyData, 1:dateCol_i, dateColWY_i, everything())
+  }
+  
+  return(wyData)
+}
+
+#' extract WY from a date
+#' 
+#' @param dateVec vector of dates as character ("YYYY-DD-MM"),
+#' Date, or POSIXct. Numeric does not work.
+#' 
+#' @keywords internal
```

Comment:
I think we should expose this. It could be useful to people. Maybe they've altered their data frame or have their own spreadsheet.